### PR TITLE
Slack systemic component

### PR DESCRIPTION
**Pull request by GentooXativa**
First version with 2 simple methods, sendMessage and sendFile

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.md]
+trim_trailing_whitespace = false
+
+[Makefile]
+indent_style = tab
```
Comment (Betisman) [inline .editorconfig:16]:
```suggestion
indent_style = tab

```

File: index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+const debug = require('debug')('systemic:slack');
+const { createReadStream, existsSync } = require('fs');
+const { WebClient } = require('@slack/web-api');
+
+module.exports = () => {
+  let slackClient;
+  const start = async ({ config }) => {
+    const token = process.env.SLACK_TOKEN;
```
Comment (feliun) [inline index.js:8]: token should come from config, env vars should be resolved by the config component

File: index.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        file: createReadStream(filepath),
+        icon_emoji: iconEmoji,
+        username: botName,
+      });
+
+      debug(`File uploaded: ${result.file.id}`);
+      return result;
+    };
+
+    return { sendMessage, sendFile };
```
Comment (feliun) [inline index.js:43]: I'm not convinced about changing the slack API, why constraining it? Systemic plugins should only provide start/stop and enforce config. If you can expose the slack API then we will save future releases to add more methods one by one, doesn't seem to provide much value

Comment (GentooXativa) [reply index.js]: I exposed the slackClient on last commit to not restrain and allow the use of whole API

File: index.js (hunk 2 above)
Comment (GentooXativa) [reply index.js]: True, updated on last commit

File: index.js (hunk 3 above)
Comment (feliun) [reply index.js]: why not?
This is the philosophy behind systemic components: https://github.com/guidesmiths/systemic-mongodb/blob/master/index.js#L31

**Comment by feliun** [comment]
Systemic components should be very very thin: https://github.com/guidesmiths/systemic-mongodb/blob/master/index.js#L31

This systemic component should purely inject a config, ensure it has everything it needs (or throw an error otherwise) and return the entire slack API

File: index.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        file: createReadStream(filepath),
+        icon_emoji: iconEmoji,
+        username: botName,
+      });
+
+      debug(`File uploaded: ${result.file.id}`);
+      return result;
+    };
+
+    return { sendMessage, sendFile, slackClient };
```
Comment (feliun) [inline index.js:43]: @GentooXativa we should purely return this

´slackClient = new WebClient(token);´

The other "helpers" should be created by the client, they don't belong to the systemic adapter. This is an un-opinionated thin layer

**Comment by GentooXativa** [comment]
@Betisman @feliun what about last changes? Are ok to merge and publish?

**Comment by feliun** [comment]
yes I think so

**Pull request merged**